inc/settings.php

```diff
@@ -60,6 +68,75 @@ function update_privacy_policy_page() {
 	update_option( $privacy_option, $updated_id );
 }
 
+/**
+ * Auto-create the policy page based on the button that was clicked.
+ *
+ * @return void
+ */
+function create_policy_page() {
+	if (
+		! isset( $_POST['create_policy_page'] ) ||
+		! in_array( esc_attr( $_POST['create_policy_page'] ), get_allowed_policy_page_values(), true )
+	) {
+		return;
+	}
+
+	$policy_page = esc_attr( $_POST['create_policy_page'] );
+
+	/**
+	 * Whether we are using the block editor.
+	 *
+	 * This defaults to true, but if false, we omit the gutenberg block support in the policy content.
+	 *
+	 * @var bool True/false whether the site is using the block editor.
+	 */
+	$block_editor = apply_filters( 'altis.consent.use_block_editor', '__return_true' );
+
+	if ( $policy_page === 'privacy_policy' ) {
+		if ( ! class_exists( 'WP_Privacy_Policy_Content' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/class-wp-privacy-policy-content.php';
+		}
+
+		$option_name         = 'wp_page_for_privacy_policy';
+		$policy_page_title   = __( 'Privacy Policy', 'altis-consent' );
+		$policy_page_content = WP_Privacy_Policy_Content::get_default_content( $block_editor );
+	} elseif ( $policy_page === 'cookie_policy' ) {
+		$option_name         = 'cookie_consent_options';
+		$policy_page_title   = __( 'Cookie Policy', 'altis-consent' );
+		$policy_page_content = CookiePolicy\get_default_content( $block_editor );
+	}
+
+	$policy_page_id = wp_insert_post( [
+		'post_title'   => $policy_page_title,
+		'post_status'  => 'publish',
```

Comment:
Default WP creates the privacy policy page as a draft. What are your thoughts around this change to default publish the content? My concern would be someone updating or adding this feature for the first time would get new content published immediately without warning or (ironically) consent. Given the legal nature of the page contents I think we should be cautious and stick to draft as the default.

Reply:
I have no problem changing it to draft. I think Complianz actually publishes it for you, but it's probably better to keep it as a draft. I think I was maybe afraid that I was filtering out anything less than published pages in the [`pages_exist` check](https://github.com/humanmade/altis-consent/blob/99aa13852abd02b6c193b6ead5dbee8f9ad69b9b/inc/settings.php#L588-L603) but that's not actually the case.